From ba3521fba7a16b385e34da91f73b4876db2d4bef Mon Sep 17 00:00:00 2001
From: Andrzej Pietrasiewicz <andrzej.p@samsung.com>
Date: Wed, 9 Oct 2013 10:08:29 +0200
Subject: [PATCH 1146/1302] usb: gadget: mass_storage: merge usb_f_mass_storage
 module with u_ms module

u_ms.ko is needed only together with usb_f_mass_storage.ko. Merge them.

Signed-off-by: Andrzej Pietrasiewicz <andrzej.p@samsung.com>
Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
Signed-off-by: Felipe Balbi <balbi@ti.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 drivers/usb/gadget/Kconfig  | 7 -------
 drivers/usb/gadget/Makefile | 4 +---
 2 files changed, 1 insertion(+), 10 deletions(-)

diff --git a/drivers/usb/gadget/Kconfig b/drivers/usb/gadget/Kconfig
index df9c040..93dfc90 100644
--- a/drivers/usb/gadget/Kconfig
+++ b/drivers/usb/gadget/Kconfig
@@ -537,9 +537,6 @@ config USB_F_SUBSET
 config USB_F_RNDIS
 	tristate
 
-config USB_U_MS
-	tristate
-
 config USB_F_MASS_STORAGE
 	tristate
 
@@ -683,7 +680,6 @@ config USB_CONFIGFS_PHONET
 config USB_CONFIGFS_MASS_STORAGE
 	boolean "Mass storage"
 	depends on USB_CONFIGFS
-	select USB_U_MS
 	select USB_F_MASS_STORAGE
 	help
 	  The Mass Storage Gadget acts as a USB Mass Storage disk drive.
@@ -907,7 +903,6 @@ config USB_MASS_STORAGE
 	tristate "Mass Storage Gadget"
 	depends on BLOCK
 	select USB_LIBCOMPOSITE
-	select USB_U_MS
 	select USB_F_MASS_STORAGE
 	help
 	  The Mass Storage Gadget acts as a USB Mass Storage disk drive.
@@ -1044,7 +1039,6 @@ config USB_G_ACM_MS
 	select USB_LIBCOMPOSITE
 	select USB_U_SERIAL
 	select USB_F_ACM
-	select USB_U_MS
 	select USB_F_MASS_STORAGE
 	help
 	  This driver provides two functions in one configuration:
@@ -1061,7 +1055,6 @@ config USB_G_MULTI
 	select USB_U_SERIAL
 	select USB_U_ETHER
 	select USB_F_ACM
-	select USB_U_MS
 	select USB_F_MASS_STORAGE
 	help
 	  The Multifunction Composite Gadget provides Ethernet (RNDIS
diff --git a/drivers/usb/gadget/Makefile b/drivers/usb/gadget/Makefile
index 62ccbda..345a49c 100644
--- a/drivers/usb/gadget/Makefile
+++ b/drivers/usb/gadget/Makefile
@@ -60,9 +60,7 @@ usb_f_ecm_subset-y		:= f_subset.o
 obj-$(CONFIG_USB_F_SUBSET)	+= usb_f_ecm_subset.o
 usb_f_rndis-y			:= f_rndis.o
 obj-$(CONFIG_USB_F_RNDIS)	+= usb_f_rndis.o
-u_ms-y				:= storage_common.o
-obj-$(CONFIG_USB_U_MS)		+= u_ms.o
-usb_f_mass_storage-y		:= f_mass_storage.o
+usb_f_mass_storage-y		:= f_mass_storage.o storage_common.o
 obj-$(CONFIG_USB_F_MASS_STORAGE)+= usb_f_mass_storage.o
 
 #
-- 
1.8.3.2

